docs/gke/iap.md

```diff
@@ -10,15 +10,58 @@ when using GKE.
 
 If you aren't familiar with IAP you might want to start by looking at those docs.
 
+The instructions below reference the following environment variables which you will need to set for your deployment
+
+  * **CORE_NAME** The name assigned to the core Kubeflow ksonnet components (this is the name chosen when you ran `ks generate...`).
+  * **ENVIRONMENT** The name of the ksonnet environment where you want to deploy Kubeflow.
+  * **FQDN** The fully qualified domain name to use for your Kubeflow deployment.
+  * **IP_NAME** The name of the GCP static IP that you created above and will be associated with **DOMAIN**.
+  * **NAMESPACE** The namespace where Kubeflow is deployed.
+  * **ACCOUNT** The email address for your ACME account where certificate expiration notifications will be sent.
+
 ### Preliminaries
 
-##### Create an external static IP address
+#### Create an external static IP address
 
 ```
+PROJECT=$(gcloud config get-value project)
 gcloud compute --project=${PROJECT} addresses create kubeflow --global
 ```
 
-Use your DNS provider (e.g. Google Domains) create a type A custom resource record that associates the host you want e.g "kubeflow"
+#### Configure DNS record
+
+[Cloud Endpoints](https://cloud.google.com/endpoints/docs/) can be used to automatically provision a free DNS record for Kubeflow in the form of: `NAME.endpoints.PROJECT.cloud.goog`. 
+
+Enable the following APIs:
+- [Google Cloud Endpoints](https://console.cloud.google.com/apis/library/endpoints.googleapis.com/?q=Cloud%20Endpoints)
+- [Google Service Management](https://console.cloud.google.com/apis/library/servicemanagement.googleapis.com/?q=Service%20Management)
+
+Create a service account an IAM bindings for the cloud-endpoints-controller:
+
+```
+gcloud iam service-accounts create cloud-endpoints-controller \
+    --display-name cloud-endpoints-controller
+export SA_EMAIL=$(gcloud iam service-accounts list \
+    --filter="displayName:cloud-endpoints-controller" \
+    --format='value(email)')
+gcloud projects add-iam-policy-binding \
+      $PROJECT --role roles/servicemanagement.admin --member serviceAccount:$SA_EMAIL
+
+gcloud iam service-accounts keys create cloudep-sa.json --iam-account $SA_EMAIL
+
+kubectl create secret generic --namespace=${NAMESPACE} cloudep-sa --from-file=./cloudep-sa.json
+```
+
+Install the `cloud-endpoints` controller and set the `FQDN` environment variable used later:
+
+```
+ks generate cloud-endpoints cloud-endpoints --namespace=${NAMESPACE} --secretName=cloudep-sa
+ks apply ${ENVIRONMENT} -c cloud-endpoints
+
+export FQDN="kubeflow.endpoints.$(gcloud config get-value project).cloud.goog"
+```
+
+Alternatively if you already have a DNS provider (e.g. Google Domains) create a type A custom resource record that associates the host you want e.g "kubeflow"
```

Comment:
If we have our own domain then we don't need a cloud-endpoints component right?

Reply:
Correct, I can separate those sections for clarity.

Reply:
The IP reservation gcloud step is also optional if you are using cloud-endpoints since the endpoint is coupled to any dynamically provisioned Ingress IP. We would just have to make the annotation on the Ingress conditional on the presence of an IP name.

Reply:
If both work lets document but if you have to reserve an IP I think that's fine. I think that's best practice anyway so that's what we'll probably recommend.